src/Aspire.Hosting/OtlpConfigurationExtensions.cs

```diff
@@ -56,8 +58,22 @@ public static void AddOtlpEnvironment(IResource resource, IConfiguration configu
             }
 
             // Set the service name and instance id to the resource name and UID. Values are injected by DCP.
-            context.EnvironmentVariables["OTEL_RESOURCE_ATTRIBUTES"] = "service.instance.id={{- .Name -}}";
-            context.EnvironmentVariables["OTEL_SERVICE_NAME"] = "{{- index .Annotations \"otel-service-name\" -}}";
+            var dcpDependencyCheckService = context.ExecutionContext?.ServiceProvider?.GetService(typeof(IDcpDependencyCheckService)) as IDcpDependencyCheckService;
+            if (dcpDependencyCheckService is null)
+            {
+                throw new InvalidOperationException("DCP dependency check service is not available"); // Should never happen.
+            }
+            var dcpInfo = await dcpDependencyCheckService.GetDcpInfoAsync(context.CancellationToken).ConfigureAwait(false);
+            if (dcpInfo?.Version?.CompareTo(DcpVersion.MinimumVersionAspire_8_1) >= 0)
```

Comment:
I don't think Version on dcp info can be null (at least, I didn't see the extra null operator in other places)
```suggestion
            if (dcpInfo?.Version.CompareTo(DcpVersion.MinimumVersionAspire_8_1) >= 0)
```

Reply:
It is defined as nullable